src/platform/tls_schannel.c

```diff
@@ -2103,6 +2108,9 @@ CxPlatTlsWriteDataToSchannel(
                     CertFreeCertificateContext(PeerCert);
                 }
 #endif
+                if ((Result & CXPLAT_TLS_RESULT_ERROR) != 0) {
```

Comment:
I think we treat fetching the client certificate as best-effort; making it a fatal error then means that applications which don't have a hard-requirement on the client certificate will always fail their connections in this case.

Reply:
Our option is either that, or we don't pass up to the user even though they've requested it. I'm not sure which is worse. Also OpenSSL would fail in this manner.

Reply:
Note that this isn't just client certificates. Server certs can be called with this too.

Reply:
Maybe it should be fatal on client for customer server cert validation (if the NO_VALIDATION flag is missing), and it should be optional on server unless the REQUIRE_CLIENT_CERTS flag is present?  That sounds complex, and would require some specific tests to ensure that code path works.

Reply:
OHHH I get what youre getting at now. Clients would fail even though its not a requirement. Thats actually not that hard to fix.

Reply:
Also, I tried to make sure that OpenSSL and Schannel behave the same in this area. So we'd have to make sure they both still act the same for this.